### ISSUE TITLE
cannot reproduce 59.6 mAP in nuScenes' readmd

### ISSUE THREAD

**Issue by poodarchu** [tianweiy/CenterPoint#367]
https://github.com/tianweiy/CenterPoint/blob/master/configs/nusc/README.md

does the performance reported here use flip aug during test?
does the link to https://github.com/tianweiy/CenterPoint/blob/master/configs/nusc/voxelnet/nusc_centerpoint_voxelnet_0075voxel_fix_bn_z.py this config match the results 59.6?

**Comment by tianweiy** [comment]
no flip; 

you can see our pretrained model / detection output at https://mitprod-my.sharepoint.com/personal/tianweiy_mit_edu/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Ftianweiy%5Fmit%5Fedu%2FDocuments%2FCenterPoint%2FNUSC%5FMODELS%2Fnusc%5Fcenterpoint%5Fvoxelnet%5F0075voxel%5Ffix%5Fbn%5Fz&ga=1 

it should match ? https://mitprod-my.sharepoint.com/:f:/g/personal/tianweiy_mit_edu/EhgzjwV2EghOnHFKyRgSadoBr2kUo7yPu52N-I3dG3c5dA?e=a9MdhX 

otherwises, I probably put some wrong file

**Comment by poodarchu** [comment]
I find that my CenterPoint re-implementation can obtain the same results when using voxel size 0.1 (56.4 mAP), but when using the same setting as your 59.6mAP config (voxel size 0.075), it can only improves about ~1 mAP.

This phenomenon can also be observed in https://github.com/open-mmlab/mmdetection3d/blob/master/configs/centerpoint/README.md

Have you ever encounter this problem?

**Comment by tianweiy** [comment]
No. 

If you look at the commit history, this 56.4 is a really old result. After fixing the batch norm bug (see https://github.com/tianweiy/CenterPoint/pull/109), the results already get 0.5~1 map better. 

After that, I added a global z axis translation augmentation in this commit, https://github.com/tianweiy/CenterPoint/commit/157a63a6a090a61bfa51965cdaeafddd8d466567 which brings the number up to 58.5 

In the end, in https://github.com/tianweiy/CenterPoint/commit/c9822e1202ff1b197db237d284f546e8cb98d766 we changed the range of scaling augmentation, which brings the final map to 59.6 

The mmdet3d is an old implementation and probably didn't have the augmentation parameters change we did in this repo. Other details may also exist. 

AFAIK, openpcdet gets reasonable close results ? (https://github.com/open-mmlab/OpenPCDet 59.2 map)

**Comment by poodarchu** [comment]
I see, the missing part should be the syncBN module in my implementation.
I'll check those stuff, thank you very much.

**Comment by poodarchu** [comment]
Is is still necessary to use SECOND's fastai optim (adam + onecycle)?
I'm actually trying to re-implement CP without legacy components like the aboves.
I find that adding SyncBN to the model has no effect to performance. Does SycnBN helps a lot in this repo?

Moreover, I find that here https://github.com/tianweiy/CenterPoint/blob/cb25e870b271fe8259e91c5d17dcd429d74abc91/det3d/builder.py#L99 the wrapper is fed into a 3e-3 lr , which is inconsistent with the lr 0.001 in the config files.

optimizer = OptimWrapper.create(
        optimizer_func,
        3e-3,
        get_layer_groups(net),
        wd=config.WD,
        true_wd=optimizer_config.FIXED_WD,
        bn_wd=True,
    )

**Comment by poodarchu** [comment]
My re-implementation of CP on the WAYMO dataset can already achieve better performance with the same setting as this repo, so I think their might be some "magic" w.r.t. the nuScenes dataset.

**Comment by tianweiy** [comment]
Is is still necessary to use SECOND's fastai optim (adam + onecycle)?
I have no clue. I use it all the same so I am not sure about other optim.

> I find that adding SyncBN to the model has no effect to performance. Does SycnBN helps a lot in this repo?
Again no comparison between syncbn and standard bn. But one thing I know is that torch's native syncbn works the same as the apex repo. 


> the wrapper is fed into a 3e-3 lr , which is inconsistent with the lr 0.001 in the config files.
Yeah, but if I remember correctly, there is lr updater at each iteration that will update the lr according to the schedule so the init value doesn't matter. 

> My re-implementation of CP on the WAYMO dataset can already achieve better performance with the same setting as this repo, so I think their might be some "magic" w.r.t. the nuScenes dataset.

Yeah, if you follow the literature, there are multiple tricks specifically for nuscenes. For instance, they do something called fading strategy where they turn off data base sampling in the last few epochs and this give a large boost. 

I am not working on 3d for a while now. So for SOTA, you probably want to check their corresponding repos for details / tricks

**Comment by poodarchu** [comment]
but cp does not use Fade Strategy to achieve 59.6mAP, right?

**Comment by tianweiy** [comment]
we didn't use fade for the 59.6 map number
